lib/jsdom/living/nodes/HTMLTableElement-impl.js

```diff
@@ -16,10 +18,60 @@ class HTMLTableElementImpl extends HTMLElementImpl {
     return firstChildWithHTMLLocalName(this, "thead");
   }
 
+  set tHead(value) {
+    if (value !== null && value._localName !== 'thead') {
+      throw new DOMException(DOMException.HIERARCHY_REQUEST_ERR,
+        "Cannot set a non-header element as a table header");
+    }
+
+    const currentHead = this.tHead;
+    if (currentHead !== null) {
+      this.removeChild(currentHead);
+    }
+
+    if (value !== null) {
+      let insertionPoint = null;
+
+      const iterator = domSymbolTree.childrenIterator(this);
+      for (const child of iterator) {
+        if (child.nodeType !== NODE_TYPE.ELEMENT_NODE) {
+          continue;
+        }
+
+        if (child._namespaceURI !== "http://www.w3.org/1999/xhtml" || (child._localName !== 'caption' && child._localName !== 'colgroup')) {
+          insertionPoint = child;
+          break;
+        }
+      }
+
+      if (insertionPoint) {
```

Comment:
I think this `if` is redundant as `insertBefore` with null second argument should already behave as you want.

Reply:
Indeed, changed.